test/date/mutt_date_parse_date.c

```diff
@@ -48,38 +48,39 @@ void test_mutt_date_parse_date(void)
-     { "Wed, 13 Jun 20 12:34:56 +0100",     1592048096 },
-     { "Wed, 13 Jun 10000 12:34:56 +0100",  -1         },
-
-     { "Wed, 13 Jun 2007 12:34 +0100",      1181734440 },
-     { "Wed, 13 Jun 2007 12.34 +0100",      -1         },
-
-     { "Wed, 13 Jun 2007 -1:34:56 +0100",   -1         },
-     { "Wed, 13 Jun 2007 24:34:56 +0100",   -1         },
-     { "Wed, 13 Jun 2007 12:-1:56 +0100",   -1         },
-     { "Wed, 13 Jun 2007 12:60:56 +0100",   -1         },
-     { "Wed, 13 Jun 2007 12:34:-1 +0100",   -1         },
-     { "Wed, 13 Jun 2007 12:34:61 +0100",   -1         },
-
-     { "Wed, 13 Jun 2007 12:34:56 -0100",  1181741696  },
-     { "Wed, 13 Jun 2007 12:34:56 -0100 (CET)",  1181741696  },
-     { "Wed, 13 Jun 2007 12:34:56 (UTC)",  1181738096  },
-     { "Wed, 13 Jun 2007 12:34:56 UTC",    1181738096  },
-
-     { "13 Jun 2007 12:34:56",             -1          }, /* TZ is mandatory */
-     { "13 Jun 2007",                      -1          },
+     { "Wed, 13 Jun 2007 12:34:56 +0100",            1181734496 },
+     /* The following timezone is a comment -> assumed UTC */
+     { "Wed, 13 Jun 2007 12:34:56 (CET)",            1181738096 },
+     /* We stop parsing the following after the time -> assumed UTC */
+     { "Wed, 13 Jun 2007 12:34:56 (CET",             1181738096},
+     { "Wed, 13 Jun 2007 12:34:56 MET DST",          1181730896 },
+     { "Wed, ab Jun 2007 12:34:56 +0100",            -1         },
+     { "Wed, -2 Jun 2007 12:34:56 +0100",            -1         },
+     { "Wed, 32 Jun 2007 12:34:56 +0100",            -1         },
+     { "Wed, 13 Bob 2007 12:34:56 +0100",            -1         },
+     { "Wed, 13 Jun -1 12:34:56 +0100",              -1         },
+     { "Wed, 13 Jun 20 12:34:56 +0100",              1592048096 },
+     { "Wed, 13 Jun 10000 12:34:56 +0100",           -1         },
+     { "Wed, 13 Jun 2007 12:34 +0100",               1181734440 },
+     { "Wed, 13 (06) Jun 2007 (seven) 12:34 +0100",  1181734440 },
+     { "Wed, 13 Jun 2007 12.34 +0100",               -1         },
+     { "Wed, 13 Jun 2007 -1:34:56 +0100",            -1         },
+     { "Wed, 13 Jun 2007 24:34:56 +0100",            -1         },
+     { "Wed, 13 (06) Jun 2007 24:34:56 +0100",       -1         },
+     { "Wed, 13 Jun 2007 12:-1:56 +0100",            -1         },
+     { "Wed, 13 Jun 2007 12:60:56 +0100",            -1         },
+     { "Wed, 13 Jun 2007 (bar baz) 12:60:56 +0100",  -1         },
+     { "Wed, 13 Jun 2007 12:34:-1 +0100",            -1         },
+     { "Wed, 13 Jun 2007 12:34:61 +0100",            -1         },
+     { "Wed, 13 Jun 2007 12:34:56 -0100",            1181741696 },
+     { "Wed, 13 Jun (Ju (06)n) 2007 12:34:56 -0100", 1181741696 },
+     { "Wed, 13 Jun 2007 12:34:56 -0100 (CET)",      1181741696 },
+     { "Wed, 13 Jun 2007 12:34:56 -0100 (CET)",      1181741696 },
+     { "Wed, 13 Jun 2007 12:34:56 (UTC)",            1181738096 },
```

Comment:
This fails (but sort of doesn't).
The timezone is regex'd as `(UTC)` which can't be parsed.

> Error: Cannot parse TZ \< (UTC)\>

but that doesn't cause `mutt_date_parse_date()` to fail.

Reply:
That's expected - that is a comment, not a proper timezone as far as I understand the RFC. I will add a comment.

Reply:
Turns out, this is already covered by https://github.com/neomutt/neomutt/pull/2242/files/6f71d0ba5db0d8fec3c49cf07d9908db6652358c#diff-ed2732a1db2886f042ef5b242f138ddcR53

Reply:
If it's a comment, then there's no timezone info (which is required), but parse still succeeds.

Reply:
you are right, I am a bit lax on the timezone and assume UTC. I will change that.